HW6/RogueLike/RogueLike/Map.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.IO;
+
+namespace RogueLike
+{
+    /// <summary>
+    /// Class for interacting with the map
+    /// </summary>
+    public class Map
+    {
+        private bool[,] mainMap;
+        private int playerPositionX;
+        private int playerPositionY;
+
+        /// <summary>
+        /// Map constructor from file
+        /// </summary>
+        public Map(string pathToMap)
+        {
+            LoadMapFromFile(pathToMap);
+        }
+
+        /// <summary>
+        /// Loading borders, map walls and character positions
+        /// </summary>
+        private void LoadMapFromFile(string pathToMapFile)
+        {
+            int lines = File.ReadAllLines(pathToMapFile).Length;
+            int columns = File.ReadAllLines(pathToMapFile)[0].Length;
+            string[] stringMap = File.ReadAllLines(pathToMapFile);
+            var map = new bool[lines, columns];
+            using FileStream fileMap = File.OpenRead(pathToMapFile);
+            int playerX = -1;
+            int playerY = -1;
+            bool flag = false;
+
+            for (int i = 0; i < lines; i++)
+            {
+                for (int j = 0; j < columns; j++)
+                {
+                    map[i, j] = stringMap[i][j] == '#' ? true : false;
```

Comment:
```suggestion
                    map[i, j] = stringMap[i][j] == '#';
```
Беда с булевой алгеброй :)